core/src/main/java/com/yahoo/ycsb/DB.java

```diff
@@ -41,97 +41,94 @@
  * target application.  For the sake of comparison between experiments we also 
  * recommend you explain the semantics you chose when presenting performance results.
  */
-public abstract class DB
-{
-	/**
-	 * Properties for configuring this DB.
-	 */
-	Properties _p=new Properties();
+public abstract class DB {
+  /**
+   * Properties for configuring this DB.
+   */
+  private Properties properties = new Properties();
+
+  /**
+   * Set the properties for this DB.
+   */
+  public void setProperties(Properties p) {
+    properties = p;
 
-	/**
-	 * Set the properties for this DB.
-	 */
-	public void setProperties(Properties p)
-	{
-		_p=p;
+  }
 
-	}
+  /**
+   * Get the set of properties for this DB.
+   */
+  public Properties getProperties() {
+    return properties;
+  }
 
-	/**
-	 * Get the set of properties for this DB.
-	 */
-	public Properties getProperties()
-	{
-		return _p; 
-	}
+  /**
+   * Initialize any state for this DB.
+   * Called once per DB instance; there is one DB instance per client thread.
+   */
+  public void init() throws DBException {
+  }
 
-	/**
-	 * Initialize any state for this DB.
-	 * Called once per DB instance; there is one DB instance per client thread.
-	 */
-	public void init() throws DBException
-	{
-	}
+  /**
+   * Cleanup any state for this DB.
+   * Called once per DB instance; there is one DB instance per client thread.
+   */
+  public void cleanup() throws DBException {
+  }
 
-	/**
-	 * Cleanup any state for this DB.
-	 * Called once per DB instance; there is one DB instance per client thread.
-	 */
-	public void cleanup() throws DBException
-	{
-	}
+  /**
+   * Read a record from the database. Each field/value pair from the result will be stored in a HashMap.
+   *
+   * @param table The name of the table
+   * @param key The record key of the record to read.
+   * @param fields The list of fields to read, or null for all of them
+   * @param result A HashMap of field/value pairs for the result
+   * @return The result of the operation.
+   */
+  public abstract Status read(String table, String key, Set<String> fields, HashMap<String, ByteIterator> result);
```

Comment:
It might be nice to change the ``HashMap`` to a ``Map`` since that shouldn't break the API and be consistent with ``Set``.